webapp/channels/src/components/drafts/draft_row.tsx

```diff
@@ -132,6 +136,16 @@ function DraftRow({
         history.push(channelUrl);
     }, [channelUrl, dispatch, history, rootId, rootPostDeleted]);
 
+    useEffect(() => {
+        // For scheduled posts, we want to show channel name.
+        // If it's a public channel, we'll be able to fetch the channel,
+        // if it's a private channel and the user is not an admin, the API call will fail and we'll
+        // show a placeholder text.
+        if (!channel && isScheduledPost) {
+            dispatch(getChannel(channelId));
```

Comment:
3/5 I'd actually prefer that we don't try to load the channel here because we should typically have all of the users' channels loaded. I don't think the extra complexity here is worth it since we already have to fall back to showing something if this request fails, and loading a single channel in this way is a bad pattern since it could lead to the app making multiple or repeated requests to load individual channels whenever the user has a scheduled post in a channel that they're not a member of.

I also wouldn't expect us to be able to send a scheduled post in a channel that the user isn't currently a member of, even if they're an admin or the channel is public, but I feel less strongly about that.

Reply:
> I'd actually prefer that we don't try to load the channel here because we should typically have all of the users' channels loaded.

This is for loading the channel user is NOT a member of and hence it is not loaded. This is to display the channel's name (if public, or if user is admin). Since we don't load channels user is not member of, wI have to load it this way, there is no other way.

> I also wouldn't expect us to be able to send a scheduled post in a channel that the user isn't currently a member of, even if they're an admin or the channel is public, but I feel less strongly about that.

That is correct, we don't send scheduled posts in a channel user is not a member of even if they are sys admins or even if the channel is public. This change is for the case when you created a scheduled post, but then left the channel or were removed. We continue displaying the scheduled posts for such channel (already discussed and agreed on) but because the user is not a channel member there, the channel isn't loaded. So I try to load it, and it will fetch the channel if the channel is public or the user is a sys admin and depending on that, I either show the channel name or show "unavailable"

Reply:
I thought I explained this in the PR description but I guess not :(

Reply:
> each time you go to this view if you have a scheduled post in a non-public channel you left

Thats incorrect. I'm storing that missing channel in redux so next time you go to the view, you won't make the API call.


> lead to multiple requests (if you go to this view with multiple scheduled posts in channels you'd left) 

Well, I can make it so that I check for all missing channels and fetch them just once before rendering the scheduled posts items. What that satisfy your concern?

Reply:
> Personally, I think it'd be fine to show them as going to No Destination if you've left the corresponding channel even if it's not ideal

Its part of UX specs to show the name. Other solutions do it as well. Unless there is no way we can do it without problems, lets stick to doing it and offering a better UX.

Reply:
@hmhealey I've made the change to add the data loader. Please re-review.